src/dds/rs-dds-option.cpp

```diff
@@ -19,15 +22,38 @@ static option_range range_from_realdds( std::shared_ptr< realdds::dds_option > c
                  dds_opt->get_stepping(),
                  dds_opt->get_default_value() };
     }
+    if( std::dynamic_pointer_cast< realdds::dds_boolean_option >( dds_opt ) )
+    {
+        return { 0, 1, 1, bool( dds_opt->get_default_value() ) ? 1.f : 0.f };
+    }
+    if( auto e = std::dynamic_pointer_cast< realdds::dds_enum_option >( dds_opt ) )
+    {
+        return { 0, float( e->get_choices().size() - 1 ), 1, (float)e->get_value_index( e->get_default_value() ) };
```

Comment:
Why do we need to cast to `float` here? We don't cast the 0 and 1

Reply:
Because C++ knows how to express constants as floats, but wants to you to state that's what you're trying to do when converting from another type...